autokeras/hypermodel/preprocessor.py

```diff
@@ -142,67 +142,174 @@ def set_weights(self, weights):
 
 
 class TextToNgramVector(base.Preprocessor):
-    """Convert raw texts to n-gram vectors."""
+    """Convert raw texts to n-gram vectors.
 
-    def __init__(self, **kwargs):
+    # Arguments
+        ngram_range: Int Tuple. Range of sizes of ngram tokens to be extracted.
+            Defaults to (1,1)
+        stop_words: Set or Iterable of strings. List of stop words to be removed
+        during tokenization
+        max_features: Positive Int. Maximum number of words to be considered during
+        tokenization
+        norm: String. Can be ('l1', 'l2' or None) Attribute to replicate
+        normalization in sklearn TfidfVectorizer
+            Defaults to 'l2'
+    """
+
+    def __init__(self,
+                 ngram_range=(1, 1),
+                 stop_words=None,
+                 max_features=None,
+                 norm='l2',
+                 ** kwargs):
         super().__init__(**kwargs)
-        self.vectorizer = text.TfidfVectorizer(
-            ngram_range=(1, 2),
-            strip_accents='unicode',
-            decode_error='replace',
-            analyzer='word',
-            min_df=2)
+        self.stop_words = stop_words
+        self.ngram_range = ngram_range
+        self.norm = norm
+        if(self.norm not in ('l1', 'l2', None)):
+            raise ValueError(
+                    "norm=%s, needs to be either 'l1','l2' or None"
+                    % self.norm)
         self.selector = None
         self.targets = None
-        self.vectorizer.max_features = const.Constant.VOCABULARY_SIZE
-        self._texts = []
-        self.shape = None
+        self._max_features = max_features or const.Constant.VOCABULARY_SIZE
+        if(self._max_features <= 0):
+            raise ValueError(
+                    "max_features=%r, needs to be a positive integer"
+                    % self._max_features)
+        self._shape = None
+        self.vocabulary = {}  # Vocabulary(Increase with the inputs)
+        self.sentence_containers = {}  # Number of sentence that contains the token
+        self.word_count = {}  # Count of words across documents
+        self._idf_vec = {}  # IDF of all the tokens
+        self.stc_num = 0  # The number of all the sentences in the raw doc
+        self.kbestfeature_value = []
```

Comment:
Use the snake case. Rename it to `k_best_feature_values`.

Reply:
renamed to k_best_idf_values to better represent behaviour